Source/Core/UICommon/DiscordPresence.cpp

```diff
@@ -89,88 +87,37 @@ void HandleDiscordJoin(const char* join_secret)
-      "G2M",  // G2ME01: Metroid Prime 2: Echoes
-      "RM3",  // RM3E01: Metroid Prime 3: Corruption
-      "R3M",  // R3ME01: Metroid Prime: Trilogy
-      "SMN",  // SMNE01: New Super Mario Bros. Wii
-      "G8M",  // G8ME01: Paper Mario: The Thousand-Year Door
-      "GPI",  // GPIE01: Pikmin (GC)
-      "R9I",  // R9IE01: Pikmin (Wii)
-      "GPV",  // GPVE01: Pikmin 2 (GC)
-      "R92",  // R92E01: Pikmin 2 (Wii)
-      "GC6",  // GC6E01: Pokemon Colosseum
-      "GXX",  // GXXE01: Pokemon XD: Gale of Darkness
-      "GBI",  // GBIE08: Resident Evil
-      "GHA",  // GHAE08: Resident Evil 2
-      "GLE",  // GLEE08: Resident Evil 3: Nemesis
-      "G4B",  // G4BE08: Resident Evil 4
-      "GSN",  // GSNE8P: Sonic Adventure 2: Battle
-      "GXS",  // GXSE8P: Sonic Adventure DX: Director's Cut
-      "SNC",  // SNCE8P: Sonic Colors
-      "G9S",  // G9SE8P: Sonic Heroes
-      "GRS",  // GRSEAF: SoulCalibur II
-      "RSL",  // RSLEAF: SoulCalibur Legends
-      "GK2",  // GK2E52: Spider-Man 2
-      "GQP",  // GQPE78: SpongeBob SquarePants: Battle for Bikini Bottom
-      "SVM",  // SVME01: Super Mario All-Stars: 25th Anniversary Edition
-      "RMG",  // RMGE01: Super Mario Galaxy
-      "SB4",  // SB4E01: Super Mario Galaxy 2
-      "G4Q",  // G4QE01: Super Mario Strikers
-      "GMS",  // GMSE01: Super Mario Sunshine
-      "GMB",  // GMBE8P: Super Monkey Ball
-      "GM2",  // GM2E8P: Super Monkey Ball 2
-      "R8P",  // R8PE01: Super Paper Mario
-      "RSB",  // RSBE01: Super Smash Bros. Brawl
-      "GAL",  // GALE01: Super Smash Bros. Melee
-      "PZL",  // PZLE01: The Legend of Zelda: Collector's Edition
-      "G4S",  // G4SE01: The Legend of Zelda: Four Swords Adventures
-      "D43",  // D43E01: The Legend of Zelda: Ocarina of Time / Master Quest
-      "SOU",  // SOUE01: The Legend of Zelda: Skyward Sword
-      "GZL",  // GZLE01: The Legend of Zelda: The Wind Waker
-      "GZ2",  // GZ2E01: The Legend of Zelda: Twilight Princess (GC)
-      "RZD",  // RZDE01: The Legend of Zelda: Twilight Princess (Wii)
-      "GHQ",  // GHQE7D: The Simpsons: Hit & Run
-      "RSP",  // RSPE01: Wii Sports
-      "RZT",  // RZTE01: Wii Sports Resort
-      "SX4",  // SX4E01: Xenoblade Chronicles
-  };
-
-  std::string region_neutral_gameid = gameid.substr(0, 3);
-  if (REGISTERED_GAMES.count(region_neutral_gameid) != 0)
-  {
-    // Discord asset keys can only be lowercase.
-    Common::ToLower(&region_neutral_gameid);
-    return "game_" + region_neutral_gameid;
-  }
-  return "";
+    std::string game_region = SConfig::GetInstance().GetGameTDBID().substr(3, 1);
+
+    std::string region_code;
+    if (game_region == "E") {
+        region_code = "US";
+    } else if (game_region == "P") {
+        region_code = "EN";
+    } else if (game_region == "J") {
+        region_code = "JA";
+    } else if (game_region == "K") {
+        region_code = "KO";
+    } else if (game_region == "C") {
+        region_code = "ZH";
```

Comment:
This is Dolphin's *third* implementation of trying to select the most appropriate GameTDB cover region code, with all three of them having their own flaws. Please try to unify the logic between here and GameFile. (The remaining implementation is in Java code written for Android, and I don't want to ask you to deal with that one.)

Reply:
Would the logic you speak of be the switch-case in `GameFile::DownloadDefaultCover()`?

Reply:
Yes.

Reply:
Should be similar as of ea500ef

Reply:
The reason I requested unifying them was because when you have several implementations of the same thing, people who want to make improvements are likely to only update one of the implementations. So, simply copying the code from there to here doesn't accomplish what I was thinking of.

Reply:
So would you like a function that both functions would use?

Reply:
Yes.

Reply:
Created a separate function in 51e1b28

Reply:
Wait something happened

Reply:
Alright should be resolved as of 67ae403